File: 백준/Bronze/2444. 별 찍기 － 7/별 찍기 － 7.py
import sys
input = sys.stdin.readline

N = int(input())

i = 1
minus = 2
cnt = 1
while cnt <= 2*N-1:
    print(" " * (N-i), end="")
    print("*" * (2*i-1), end="")
    print(" ")

    if cnt >= N:
        i = cnt+1 - minus
        minus += 2
    else:
        i += 1

    cnt += 1

File: 백준/Bronze/2444. 별 찍기 － 7/별 찍기 － 7.js
let fs = require('fs')
const input = fs.readFileSync("/dev/stdin").toString().trim()
// const input = fs.readFileSync("./EX.txt").toString().trim()

let N = Number(input)

let i = 1
let minus = 2
let cnt = 1
let answer = ""

while (cnt <= 2*N - 1) {
    answer += " ".repeat(N-i) + "*".repeat(2*i-1) + "\n"
    
    if (cnt >= N) {
        i = cnt + 1 - minus
        minus += 2
    }
    else {
        i += 1
    }
    cnt += 1
}

console.log(answer)